src/cil.ml

```diff
@@ -5044,7 +5069,7 @@ let makeVarinfo global name ?init typ =
     { vname = name;
       vid   = newVID ();
       vglob = global;
-      vtype = if global then typ else typeRemoveAttributes ["const"] typ;
+      vtype = if global then typ else typeRemoveAttributes ["pconst"] typ;
```

Comment:
Is it ok to only ignore pconst here? Or do we need to ignore const as well?

Reply:
Removing "const" is exactly what we don't want to do here, because then we screw up some type comparisons where it is needed for correctness.